File: java/quarkus/gelf/src/main/java/org/acme/GreetingResource.java
package org.acme;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/hello")
public class GreetingResource {

    private static final Logger LOGGER = LoggerFactory.getLogger(GreetingResource.class);

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        LOGGER.info("Hello executed");
        return "Hello RESTEasy";
    }
}

File: java/quarkus/datasource-setup/src/main/kotlin/org/acme/GreetingResource.kt
package org.acme

import io.agroal.api.AgroalDataSource
import io.quarkus.agroal.DataSource
import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType.TEXT_PLAIN

@Path("/hello")
class GreetingResource {

    @Inject
    @DataSource("mysql")
    lateinit var mySqlDataSource: AgroalDataSource

    @Inject
    @DataSource("h2")
    lateinit var h2DataSource: AgroalDataSource

    @GET
    @Produces(TEXT_PLAIN)
    fun hello(): String {
        val call = mySqlDataSource.connection.prepareCall("select * from greeting")
        call.execute()
        val resultSet = call.resultSet
        resultSet.next()
        resultSet.next()
        return resultSet.getString(2)
    }
}
